.gitignore

```diff
@@ -17,4 +17,5 @@ node_modules
 .DS_Store 
 config/*
 logs/
-.coverage
\ No newline at end of file
+.coverage
+.devcontainer/cbpi-dev-config/*
```

Comment:
The idea behind the cbpi-dev-config folder was, that a ready to go dev environment including common use cases can be provided. This should help to get developers up quickly without needing to setup a entire configuration from scratch when just wanting to get started or review a common scenario. 

When ignoring this folder this is no longer possible. Instead of ignoring it it might be better to provide a valid config that can be used in a better way as it is now (Some sensors/actors? Sample dashboard? Sample hardware that use the sensors? Something else?).

Reply:
Hey, thanks for the feedback,

My intention was to only commit file changes in the config folder if forced with `git add -f` but to ignore changes when using `git add .` . So not every commit spam changes the dev config. But this only works for new files and not for files that are already tracked. Still i think this should stay like that. When we want to commit a working complete config or parts of it we can just do it with `git add -f`. But there wont be any accidentaly commits of additional files in this folder.

What we could do instead: remove everything inside the dev config folder and just put our dev config inside a restored_config.zip file that we force commit. Then when cloning the dev config is extracted but all the new files are not in any commit because they are ignored by the .gitignore. (this would only mark the restored_config.zip as deleted ... dont know what to do with that)

please let me know if i should revert this change.

Reply:
Thanks for the clarification. Sounds good to me. I always like when things like this are obvious to others, maybe we can improve on that: 

- You could add a comment in the gitignore stating that the folder is ignored to keep its clean state and avoid accidentally updating. If updating is intended the dev should execute the step described in the next bullet point
- Create another vscode task "Update development environment config" or similar, that takes the current config and adds all changes and new files under `.devcontainer/cbpi-dev-config` as you described.

I'm not a fan of the zip backup because it hides what was changed and for what reason, because the contents of the zip file are just seen as binary by the git history.

What do you think?

Reply:
Once a file like `config.yaml` is committed (tracked) any changes will be added by `git add .` even if the file is mentioned in the .gitignore file. This is exactly what i misunderstood on my first commit.

I agree that its a bad thing to hide changes inside the zip.

But i like working with vscode tasks. maybe we can commit a dev config in a seperate folder and then use vscode tasks to load this config and overwrite the current dev config or to overwrite the comitted dev config with the current config so it can be commited. Then no files in the actual dev config folder would be commited ever.

Reply:
:+1: